.github/workflows/delploy_cdk_stack.yml

```
@@ -0,0 +1,86 @@
+name: Deploy cdk stack
+
+permissions:
+  id-token: write
+  contents: read
+
+on:
+  workflow_call:
+    inputs:
+      stage:
+        required: true
+        type: string
+    secrets:
+      target-account:
+        required: true
+
+jobs:
+  deploy-cdk-stack:
+    runs-on: ubuntu-latest
+    needs:
+      - build-layer
+    strategy:
+      fail-fast: false
+      matrix:
+        region: [
+          "af-south-1",
+          #          "eu-central-1",
+          #          "us-east-1",
+          #          "us-east-2",
+          #          "us-west-1",
+          #          "us-west-2",
+          #          "ap-east-1",
+          #          "ap-south-1",
+          #          "ap-northeast-1",
+          #          "ap-northeast-2",
+          #          "ap-southeast-1",
+          #          "ap-southeast-2",
+          #          "ca-central-1",
+          #          "eu-west-1",
+          #          "eu-west-2",
+          #          "eu-west-3",
+          #          "eu-south-1",
+          #          "eu-north-1",
+          #          "sa-east-1",
+          #          "ap-southeast-3",
+          #          "ap-northeast-3",
+          #          "me-south-1"
+        ]
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: aws credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: ${{ matrix.region }}
+          role-to-assume: arn:aws:iam::${{ secrets.target-account }}:role/${{ secrets.AWS_GITHUB_OIDC_ROLE }}
+      - name: Print assumed role
+        run: aws sts get-caller-identity
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16.12'
+          cache: 'npm'
+      - name: Setup python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+          cache: 'pip'
+      - name: install cdk and deps
+        run: |
+          npm install -g aws-cdk@2.29.0
+          cdk --version
+      - name: install deps
+        run: |
+          cd layer
+          pip install -r requirements.txt
```

nit

```suggestion
        run: pip install -r layer/requirements.txt
```

Should we also install `layer/requirements-dev.txt`?

---

`requirements-dev.txt` is not ... required 😅

---

Hmm the weird part is that everything needs to run inside the `layer` folder, since it's an isolated project, therefore change dir to `layer` and run everything else there, including cdk commands, and unzip.

---

yeah it's just that I didn't see it being used, so I thought: "Hmmm? Lemme ask Alex"